modin/pandas/test/internals/test_benchmark_mode.py

```diff
@@ -15,17 +15,38 @@
 
 from contextlib import nullcontext
 import modin.pandas as pd
-from modin.pandas.test.utils import test_data_values
+from modin.pandas.test.utils import test_data_values, df_equals
 from modin.config import BenchmarkMode, StorageFormat
 from modin.test.test_utils import warns_that_defaulting_to_pandas
 
 
+default_to_pandas_context = (
+    warns_that_defaulting_to_pandas()
+    if StorageFormat.get() == "Omnisci"
+    else nullcontext()
+)
+
+
 def test_syncronous_mode():
     assert BenchmarkMode.get()
     # On Omnisci storage, transpose() defaults to Pandas.
-    with (
-        warns_that_defaulting_to_pandas()
-        if StorageFormat.get() == "Omnisci"
-        else nullcontext()
-    ):
+    with default_to_pandas_context:
         pd.DataFrame(test_data_values[0]).mean()
+
+
+def test_serialization():
```

Comment:
the test doesn't explicitly follow the repro from #4273 but still triggers the same scenario: the original issue occurred due to serialization of Series argument when sending it between workers (so basically doing `series.__reduce__`), which triggered to execute `series.finalize()` that failed in benchmark mode.

the original reproducer couldn't be applied as `Series.isin` itself doesn't work at all with Dask engine (doesn't matter what's the value of `BenchmarkMode` variable is); raised a separate issue for this #4432

Reply:
Why not just explicitly call `finalize` function?

Reply:
added explicit `finalize` test as well